atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -549,125 +552,79 @@ public void testTransactionWriteWriteConflicts() throws Exception {
         t2.commit();
     }
 
-    @Test(expected = TransactionFailedRetriableException.class)
-    public void testValidateExternalAndCommitLocksForGet() throws Exception {
-        testValidateExternalAndCommitLocks(
-                new LockAwareTransactionTask<Void, Exception>() {
-                        @Override
-                        public Void execute(Transaction t, Iterable<HeldLocksToken> heldLocks)
-                                throws Exception {
-                            t.get(TABLE_SWEPT_THOROUGH, ImmutableSet.of(Cell.create("row1".getBytes(),
-                                            "column1".getBytes())));
-                            return null;
-                        }
-                    });
+    @Test
+    public void readsFromThoroughlySweptTableShouldFailWhenLocksAreInvalid() throws Exception {
+        for (Pair<String, LockAwareTransactionTask<Void, Exception>> task : getThoroughTableReadTasks()) {
+            try {
+                runTaskWithInvalidLocks(task.getRight());
+                Assert.fail(String.format(
```

Comment:
This will stop execution at the first failing test, which you would then have to fix to find out about remaining failures.

It would be preferable to collect `task.getLeft()` for the tasks which wrongly succeeded, and then do one assert after the for loop to show all failures at once.